### PR TITLE
Update HOC tiles on /courses and /sign_in

### PR DESCRIPTION
**Pull request by maddiedierker**
Updates the course tiles on `/courses` and `/users/sign_in`.

Some examples of those with the new Dance Party tile:

On `/users/sign_in`:
screen shot 2018-11-08 at 3 02 54 pm

On `/courses`:
screen shot 2018-11-08 at 3 02 36 pm

For both of these pages, users with different languages settings should see the following course tiles when the `DCDO.hoc_launch` flag is set to different values:

--|"mc"|"dance"|nil
--|--|--|--
**English users**|minecraft aquatic, starwars, frozen, classic maze|dance party, minecraft aquatic, _app lab, flappy code_|minecraft hero, starwars, frozen, classic maze
**non-English users**|[same as English users]|dance party, minecraft aquatic, _frozen, classic maze_|[same as English users]

### CODE REVIEW

**Comment by maddiedierker** [comment]
[HOC Dance Party tile updates](https://trello.com/c/nYRZwSsv/7-hoc-dance-party-tile-updates)

**Comment by maddiedierker** [comment]
@poorvasingal can you take a look at the table i wrote out in the description of this PR to make sure it's correct? i laid out which course tiles English/non-English users should see based on which `DCDO.hoc_launch` value is set and want to make sure i didn't get any mixed up. thank you!

**Comment by Erin007** [comment]
The Minecraft text is a little squished with "problem" on the right hand side. Does it need a little padding?
screen shot 2018-11-08 at 7 35 39 pm

Also intentional that the /courses boxes have the green "Try now" band on the bottom, but the sign in ones don't. Just curious about the consistency.

File: apps/src/templates/studioHomepages/CourseBlocks.jsx [hunk 1]
```diff
@@ -134,15 +134,33 @@ class CourseBlocksCsfLegacy extends Component {
...
   componentDidMount() {
-    const minecraftElement = this.props.displayMinecraftAquatic ? '#aquatic' : '#hero';
-    $(minecraftElement).appendTo(ReactDOM.findDOMNode(this.refs.minecraft)).show();
-    $('#starwars').appendTo(ReactDOM.findDOMNode(this.refs.starwars)).show();
-    $('#frozen').appendTo(ReactDOM.findDOMNode(this.refs.frozen)).show();
-    $('#hourofcode').appendTo(ReactDOM.findDOMNode(this.refs.hourofcode)).show();
+    // First row, dynamically created based on hocLaunch value
+    const tiles = this.getFirstRowTiles();
+    tiles.forEach((tile, index) => {
+      $(tile).appendTo(ReactDOM.findDOMNode(this.refs[index]));
```
Comment (Erin007) [inline apps/src/templates/studioHomepages/CourseBlocks.jsx:160]: snazzy.  I like how you consolidated these!

File: apps/src/templates/studioHomepages/CourseBlocks.jsx [hunk 2]
```diff
@@ -153,17 +171,19 @@ export class CourseBlocksHoc extends Component {
...
+          <ProtectedStatefulDiv ref="3"/>
         </div>
 
         {this.props.rowCount > 1 && (
           <div>
             <br/>
             <br/>
             <div className="row">
+              {/* TODO: (madelynkasula) If Flappy is in 1st row, what should replace Flappy on line below? */}
+              {/* Can we remove 2nd row? It's never used... */}
```
Comment (Erin007) [inline apps/src/templates/studioHomepages/CourseBlocks.jsx:186]: Any resolution to this yet? I'm curious.

File: apps/src/templates/studioHomepages/CourseBlocks.jsx [hunk 3]
```diff
@@ -180,7 +200,7 @@ export class CourseBlocksAll extends Component {
   static propTypes = {
     isEnglish: PropTypes.bool.isRequired,
     showModernElementaryCourses: PropTypes.bool.isRequired,
-    displayMinecraftAquatic: PropTypes.bool,
+    hocLaunch: PropTypes.bool,
```
Comment (Erin007) [inline apps/src/templates/studioHomepages/CourseBlocks.jsx]: string?

**Comment by poorvasingal** [comment]
Hey Maddie - Double checked the comments from Marina on the spec and your table looks right based on that. The main thing I'm unsure about is if the DCDO flag is going to be changed back to "nil" after HOC? B/c we'd want the tiles you have for the dance launch to continue to stay post-HOC. 

+1 to ErinB's comment on padding for the Minecraft tile. Seems like we need additional margin on the right in the tiles. I think this is true of each tile, just that it's only noticeable on MC in English b/c of the way the text ends up wrapping. 

ErinB - The "Try now" inconsistency is as expected. We only show the "Try now" band when you're signed in (even on the courses page). It's because it turns into "Continue" if you've already made progress in the tutorial, which we only know for a user if they are signed in.

File: apps/src/templates/studioHomepages/CourseBlocks.jsx (hunk 3 above)
Comment (maddiedierker) [reply apps/src/templates/studioHomepages/CourseBlocks.jsx]: 😱 yes, good catch!

File: apps/src/templates/studioHomepages/CourseBlocks.jsx (hunk 2 above)
Comment (maddiedierker) [reply apps/src/templates/studioHomepages/CourseBlocks.jsx:186]: no, i totally forgot--thank you for reminding me! 😂 i'll talk to poorva and address in a follow-up

**Pull request merged**